hottbox/algorithms/decomposition/cpd.py

```diff
@@ -163,7 +163,10 @@ def decompose(self, tensor, rank, keep_meta=0, kr_reverse=False):
             1 - keep only mode names
             2 - keep mode names and indices
         kr_reverse : bool
-
+        factor_mat : list(np.ndarray)
+            Initial list of factor matrices. 
+            Specifying this option will ignore ``init``.
```

Comment:
One thing to keep in mind. If we specify `factor_mat` and we skip `self._init_fmat()` then `self.cost` won't be reset for the `cpd.decompose`.

Reply:
Good spot. For now I have added self.cost = [] in decompose, but is there anything wrong with removing it entirely from _init_fmat and putting it on the top of decompose?

Reply:
Yeah, agree. Don't remember why I had it in `self._init_fmat()` in the first place.

Reply:
Changed implementation and test now